src/components/UserProfile/AssignBadgePopup.jsx

```diff
@@ -31,25 +32,35 @@ const AssignBadgePopup = (props) => {
     props.handleSubmit();
     props.close();
   };
+  useEffect(() => {
+    loadAllBadges();
+  }, []);
 
-  const filterBadges = (allBadges) => {
-    let filteredList = allBadges.filter((badge) => {
+  const loadAllBadges = async () => {
+    try {
```

Comment:
code looks clean.

Reply:
<img width="1910" alt="Screenshot 2023-01-06 at 11 46 29 PM" src="https://user-images.githubusercontent.com/67836976/211140135-a6837b84-2a5c-447e-829b-d99e9e2732e9.png">

Reply:
This is the error I see after assigning the badges.

Reply:
Hi guys,  I tested my code here again, and I am not getting this error, take a look at the video that I did

https://user-images.githubusercontent.com/39320057/211143301-75e5d06e-e85b-43b9-ba0e-46dc46e38a8b.mp4